### PR TITLE
TASK-1: add simple BarChart

### PR DESCRIPTION
**Pull request by Ktyby**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,16 +27,27 @@
...
-		"build"
-	]
+		"build",
+		"lint"
+	],
+	"dependencies": {
+		"prop-types": "^15.7.2",
+		"react": "^17.0.2",
+		"react-dom": "^17.0.2",
+		"stylelint-config-recommended": "^5.0.0"
```
Comment (Ktyby) [inline package.json]: replace it to devdependencies

File: src/components/App/App.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import BarChart from '../BarChart';
+
+import './style.css';
+
+const App = () => (
+	<div className="chart">
+		<BarChart />
```
Comment (MaksimLysakou) [inline src/components/App/App.jsx]: Do not use instance here.
Use storybook to render a chart

File: src/components/BarChart/Axes/Axes.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		{intervals.map((element, index) => (
+			<div className="chart__interval" key={index}>
+				{element}
+			</div>
+		))}
+	</div>
+));
+
+Axes.propTypes = {
+	intervals: PropTypes.arrayOf(PropTypes.string).isRequired,
```
Comment (MaksimLysakou) [inline src/components/BarChart/Axes/Axes.jsx]: A single axis "number with line" is called "tick".

The property is better to name "domain"
"Domain" - is a range of values for the axis. [min, max] - for numeric axis. [1st, 2ns, 3rd...] - for named axis.
For your first task, let's use just named axis.

File: src/components/BarChart/Axes/Axes.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React, { memo } from 'react';
+import PropTypes from 'prop-types';
+
+import './style.css';
+
+const Axes = memo(({ intervals }) => (
```
Comment (MaksimLysakou) [inline src/components/BarChart/Axes/Axes.jsx]: Axes - plural
Axis - singular.

"BottomAxis" - is the best name for the component

File: src/components/BarChart/Bars/hooks/useSetBarHeight.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+const useSetBarHeight = (parentElement) => {
+	useEffect(() => {
+		const bars = document.querySelectorAll('.charts__bar');
+		const maxHeight = parentElement.current.clientHeight;
+
+		bars.forEach((element) => {
+			const barValue = +element.getAttribute('value');
+			const barHeight = (maxHeight * barValue) / 100;
+
+			element.style.height = `${barHeight}px`;
```
Comment (MaksimLysakou) [inline src/components/BarChart/Bars/hooks/useSetBarHeight.js]: Why do you use px?
Using percents will be a mush simple way.
You don't need to know a parent's height for percents.
And it's okay to set this login into Bars component

File: src/components/BarChart/utils/getMeasureArray.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+const getMeasureArray = (data, measure) => (measure === 'axis' ? data.map(({ x }) => x) : data.map(({ y }) => y));
```
Comment (MaksimLysakou) [inline src/components/BarChart/utils/getMeasureArray.js]: I see no sense in this function. We can do it in component

File: src/components/BarChart/BarChart.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import React, { useRef } from 'react';
+
+import Axes from './Axes';
+import Bars from './Bars';
+
+import { getMeasureArray } from './utils';
+
+import './style.css';
+
+const chartsData = [
```
Comment (MaksimLysakou) [inline src/components/BarChart/BarChart.jsx]: It should be passed as props

Comment (Ktyby) [reply src/components/BarChart/BarChart.jsx]: Where from?

Comment (Kiryl-Henets) [reply src/components/BarChart/BarChart.jsx]: You can pass any props to component using storybook

File: src/components/BarChart/Bars/hooks/useSetBarHeight.js (hunk 5 above)
Comment (Kiryl-Henets) [reply src/components/BarChart/Bars/hooks/useSetBarHeight.js]: Your calculations also shouldn't depend on the values from DOM usually. You have the data about bar value in the chartData, so there is no need to take it from the attribute, as you do

**Review by MaksimLysakou — APPROVED** [review]

**Pull request merged**